teps/0111-remove-volumes-from-task-spec.md

```diff
@@ -0,0 +1,252 @@
+from the host node.
+No image build Tasks in the Catalog mount files from host nodes.
+
+### Use cases not currently supported by Workspaces
+
+#### Docker in Docker (advanced use cases)
+
+Some users would like to be able to access the host node's image cache for performing Docker builds.
+This requires mounting the host's
+[docker socket](https://docs.docker.com/engine/reference/commandline/dockerd/#daemon-socket-option).
+The most natural option to do so is a [hostPath volume](https://kubernetes.io/docs/concepts/storage/volumes/#hostpath).
+Workspaces do not support hostPath volumes as bindings, so this feature would not be supported
+by Workspaces. However, this approach is not recommended for
+[security reasons](https://blog.quarkslab.com/kubernetes-and-hostpath-a-love-hate-relationship.html).
+
+The only Catalog Task using hostPath volumes is [`kind`](https://hub.tekton.dev/tekton/task/kind).
+For more information on why `kind` on Kubernetes requires hostPath volumes, see
+https://github.com/kubernetes-sigs/kind/issues/303#issuecomment-521384993 and
+[Running KIND Inside A Kubernetes Cluster For Continuous Integration](https://d2iq.com/blog/running-kind-inside-a-kubernetes-cluster-for-continuous-integration)
+(TL;DR: because `kind` needs to create nested Docker containers).
+
+#### Tasks that require block devices
+
+[Block devices](https://kubernetes.io/blog/2019/03/07/raw-block-volume-support-to-beta/) support random access to data in
+fixed-size blocks. The most common use case for block devices is in database implementations,
+which is not very relevant for the CI/CD use cases Tekton aims to support.
+
+One CI/CD use case for block devices in Tasks is that [some image builds require them](https://github.com/tektoncd/pipeline/issues/1438#issuecomment-544313339).
+However, we have received only one request for support for block devices in workspaces, which may indicate that not many
+users need this feature. (It could also mean that people are happily using `volumeDevices` in `Steps` or that they just
+did not see the linked issue.) There are no catalog Tasks using this feature.
+
+Some CSI drivers support block devices (see 
+[in-tree support for volume types](#in-tree-support-for-many-types-of-volumes-in-workspace-bindings) for more info).
+
+## Proposal
+
+1. Allow workspaces to support the common CI/CD use cases Tekton targets, especially Docker-in-Docker.
+  - Promote workspaces in sidecars and isolated workspaces to beta.
+  - Optionally, add support for [CSI workspace bindings](#in-tree-support-for-many-types-of-volumes-in-workspace-bindings).
+1. Update examples our documentation to demonstrate the use of workspaces instead of volumes for the use cases listed in
+this TEP.
+  - The v1beta1 -> v1 migration guide should include examples of migrating volumes to workspaces.
+1. Deprecate `Task.Spec.Volumes`, `Step.VolumeMounts`, and `Step.VolumeDevices` (and likewise for `StepTemplate` and `Sidecar`).
```

Comment:
We have another use case not already covered here and not supported by workspaces (nor could they easily be) which is dynamic volumes. Sometimes, it is desired/opportune to create a volume (say backed by a configmap or secret) dynamically in a pipeline and then mount it into an already named volume mount in a task. This is not possible with workspaces because the workspace bindings need to be defined up front.

Reply:
Thanks Pierre! I added this as a use case that workspaces don't currently support, and modified the TEP to specify that before it's merged as implementable, we should have a plan for workspaces support for this use case. PTAL!